### PR TITLE
Add the ability to adjust the line separator based on the system.

### PR DESCRIPTION
**Pull request by the-end-sea-of-the-world**
Deal with issue [#600](https://github.com/square/javapoet/issues/600). Check the system and use the corresponding line separator.

### CODE REVIEW

File: src/main/java/com/squareup/javapoet/JavaFile.java [hunk 1]
```diff
@@ -333,4 +327,31 @@ public JavaFile build() {
...
+      LS = "\r\n";
+    }else if (osName.contains("Linux")){
+      LS = "\n";
+    }else if (osName.contains("Mac OS X")){
+      LS = "\n";
+    }else if (osName.contains("Mac OS")){
+      LS = "\r";
+    }else{
+      LS = "\n";
+    }
```
Comment (ppkarwasz) [inline src/main/java/com/squareup/javapoet/JavaFile.java:345]: I would use `System.lineSeparator()` instead.

File: src/main/java/com/squareup/javapoet/AnnotationSpec.java [hunk 2]
```diff
@@ -53,8 +53,8 @@ private AnnotationSpec(Builder builder) {
   }
 
   void emit(CodeWriter codeWriter, boolean inline) throws IOException {
-    String whitespace = inline ? "" : "\n";
-    String memberSeparator = inline ? ", " : ",\n";
+    String whitespace = inline ? "" : JavaFile.addLineSeparator("");
+    String memberSeparator = inline ? ", " : JavaFile.addLineSeparator(",");
```
Comment (ppkarwasz) [inline src/main/java/com/squareup/javapoet/AnnotationSpec.java:57]: What about a `JavaFile.addLineSeparator(String src, boolean inline)` helper function instead?

**Review by ppkarwasz — COMMENTED** [review]
Looks good to me, any chance seeing this in the next release?

**Comment by JakeWharton** [comment]
This would break many things. For one, you could no longer easily test against golden strings (something this library itself relies on). But even worse is that you break distributed caching of tasks in libraries like Bazel and Gradle which will compute different hashes due to line ending changes despite the contents otherwise being the same.

**Pull request closed without merging**